dashboard/modules/job/job_head.py

```diff
@@ -56,9 +71,45 @@ async def _parse_and_validate_request(
                 status=aiohttp.web.HTTPBadRequest.status_code,
             )
 
-    def job_exists(self, job_id: str) -> bool:
-        status = self._job_manager.get_job_status(job_id)
-        return status is not None
+    async def find_job_by_ids(self, job_or_submission_id: str) -> Optional[JobDetails]:
+        """
+        Attempts to find the job with a given submission_id or job id.
+        """
+        # First try to find by job_id
+        driver_jobs, submission_job_drivers = await self._get_driver_jobs()
+        job = driver_jobs.get(job_or_submission_id)
+        if job:
+            return job
+        # Try to find a driver with the given id
+        submission_id = next(
+            (
+                id
+                for id, driver in submission_job_drivers.items()
+                if driver.id == job_or_submission_id
+            ),
+            None,
+        )
+
+        if not submission_id:
+            # If we didn't find a driver with the given id,
+            # then lets try to search for a submission with given id
+            submission_id = job_or_submission_id
+
+        job_info = await asyncio.get_event_loop().run_in_executor(
+            self._executor, lambda: self._job_manager.get_job_info(submission_id)
+        )
+        if job_info:
+            driver = submission_job_drivers.get(submission_id)
+            job = JobDetails(
+                **dataclasses.asdict(job_info),
+                submission_id=submission_id,
+                job_id=driver.id if driver else None,
+                driver_info=driver,
+                type=JobType.SUBMISSION,
+            )
+            return job
+
+        return None
 
     @routes.get("/api/version")
```

Comment:
Sorry for not seeing this earlier, it looks like we can use this endpoint in the future to make breaking API changes smoother for third-party job managers that use our API.

Reply:
ill bump this